examples/job/work-queue-1/README.md

```diff
@@ -0,0 +1,339 @@
+
+In practice, if there is large data that is needed in a read-only mode by all pods
+of the Job, you will typically put that in a shared file system like NFS and mount
+that readonly on all the pods, or the program in the pod will natively read data from
+a cluster file system like HDFS.
+
+For our example, we will create the queue and fill it using the amqp command line tools.
+In practice, you might write a program to fill the queue using an amqp client library.
+
+```console
+
+$ /usr/bin/amqp-declare-queue --url=$BROKER_URL -q job1  -d
+job1
+$ for f in apple banana cherry date fig grape lemon melon
+do
+  /usr/bin/amqp-publish --url=$BROKER_URL -r job1 -p -b $f
+done
+```
+
+So, we filled the queue with 8 messages.
+
+## Create an Image
+
+Now we are ready to create an image that we will run.
+
+We will use the `amqp-consume` utility to read the message
+from the queue and run our actual program.  This has the advantage that you
+do not need to modify your program to be aware of the queue.
+So, we define a Dockerfile for an image that contains our main program,
+and the `amqp-tools`.
+
+```console
+$ mkdir tmp
+$ cd tmp
+$ cat > worker.py <<EOF
+#! /usr/bin/python
+import sys
+print("Processing " + sys.stdin.lines())
+EOF
+$ chmod +x worker.py
+$ cat > Dockerfile <<EOF
+# Specify BROKER_URL and QUEUE when running
+FROM ubuntu:14.04
+
+RUN apt-get update && apt-get install -y curl ca-certificates amqp-tools python
+--no-install-recommends && rm -rf /var/lib/apt/lists/*
+COPY ./worker.py /worker.py
+
+CMD  /usr/bin/amqp-consume --url=$BROKER_URL -q $QUEUE -c 1 /worker.py
+EOF
+$ cd ..
+$ docker build -t job-wq-1 ./tmp
+```
+
+For the [Docker Hub](https://hub.docker.com/), tag your app image with
+your username and push to the Hub with the below commands. Replace
+`<username>` with your Hub username.
+
+```
+docker tag job-wq-1 <username>/job-wq-1
+docker push <username>/job-wq-1
+```
+
+If you are using [Google Container
+Registry](https://cloud.google.com/tools/container-registry/), tag
+your app image with your project ID, and push to GCR. Replace
+`<project>` with your project ID.
+
+```
+docker tag job-wq-1 gcr.io/<project>/job-wq-1
+gcloud docker push gcr.io/<project>/job-wq-1
+```
+
+## Defining a Job
+
+Here is the job definition:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Job
+metadata:
+  name: job-wq-1
+spec:
+  selector:
+    app: job-wq-1
+  completions: 8
+  parallelism: 2
+  template:
+    metadata:
+      name: job-wq-1
+      labels:
+        app: job-wq-1
+    spec:
+      containers:
+      - name: c
+        image: gcr.io/<project>/job-wq-1
+      restartPolicy: Never
+```
+
+In this example, each pod works on one item from the queue and then exits.
+So, the completion count of the Job corresponds to the number of work items
+done.  So we set, `.spec.completions: 8` for the example, since we put 8 items in the queue.
+
+We set the `.spec.template.spec.restartPolicy` to `Never`.  You could also set it to `OnFailure`.
```

Comment:
Why? Explain in more detail or at least link to where restart policy is discussed.